tests/integration/targets/latest/tasks/dynamic_group.yml

```diff
@@ -0,0 +1,170 @@
+---
+##
+##
+### PYNAUTOBOT_DYNAMIC_GROUP
+##
+##
+
+- name: "NAUTOBOT 2.3+ DYNAMIC GROUPS TESTS"
+  when: 
+    - "nautobot_version is version('2.3', '>=')"
```

Comment:
Dynamic groups definitely existed before 2.3. Is there something specific we are version gating here?

Reply:
Some things like tags and tenants were added in 2.3 which are being tested and would fail in previous versions. 

https://docs.nautobot.com/projects/core/en/stable/user-guide/platform-functionality/dynamicgroup/#creating-dynamic-groups-in-the-ui

Reply:
Ok, in that case we can use a pattern like the one below to omit the option in 2.3+ so the rest of the module can be tested for lower versions.

```
tenant: "{{'Test Tenant' if nautobot_version is version('2.3', '>=') else omit }}"
```

Reply:
One other thing to note, the `choices` include `static` which is only a 2.3+ feature.  It could be misleading if you're wanting to separate 2.3- and 2.3+ features.

Reply:
In that case we can add it as a note in the description.

```
    description:
      - Required if I(state=present) and the dynamic group does not exist yet
      - Note: C(static) is only a valid choice in Nautobot 2.3+
    choices: [ dynamic-filter, dynamic-set, static ]
```

We don't need to necessarily test every permutation of every option, so we can write the test to use a choice that is more universal to ensure that the `group_type` option works on all versions.

Reply:
> Ok, in that case we can use a pattern like the one below to omit the option in 2.3+ so the rest of the module can be tested for lower versions.

Which module has an example of this pattern?   Curious how the assertions look following the task.

Reply:
Location type, but we just skip the assertion for that field.